Update to use version 2.0 of social client packages

epihacker opened the pull request:
Updated the social client packages to version 2.0 and updated the code to conform to that version.

cbanner commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Ratings/PageRatingRepository.cs:
```diff
@@ -125,31 +126,31 @@ public void AddRating(string user, string target, int value)
                         result = new PageRatingStatistics
                         {
-                            Average = (double)statistics.Sum/statistics.TotalCount,
+                            Average = (double)statistics.Sum / statistics.TotalCount,
```
We can probably switch this guy over to just `statistics.Mean`.

cbanner commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/ActivityStreams/CommunityFeedRepository.cs:
```diff
@@ -77,11 +74,9 @@ public IEnumerable<CommunityFeedItemViewModel> Get(CommunityFeedFilter filter)
         }
 
-        private IEnumerable<CommunityFeedItemViewModel> AdaptFeedItems(List<Composite<FeedItem, CommunityActivity>> feedItems)
+        private IEnumerable<CommunityFeedItemViewModel> AdaptFeedItems(List<FeedItem<CommunityActivity>> feedItems)
```
The adapter used in this method is one of the key areas where changes will need to be made regarding the handling of activity data on feed items.

Previously, SocialAlloy leveraged a base CommunityActivity class that was extended by a CommentActivity and RatingActivity. It then used a visitor to handle the different types uniquely. It may be easiest to just settle on a common format for simply a CommunityActivity. That will streamline things and eliminate the adaptation complexity of trying to work with multiple different types at once.

epihacker replied:
btw, all this was reworked.  Let me know if all is good with the new approach.

epihacker commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Groups/CommunityMemberRepository.cs:
```diff
@@ -100,41 +96,37 @@ public IEnumerable<CommunityMember> Get(CommunityMemberFilter communityMemberFil
+            {
+                PageInfo = new PageInfo { PageSize = communityMemberFilter.PageSize },
+                OrderBy = new List<SortInfo> { new SortInfo(MemberSortFields.Id, false) },
+                Filter = filter
```
is the following equivalent:

            this.memberFilters.Extension.Type.Is<MemberExtensionData>();

cbanner replied:
`.Is<T>()` will filter by .NET type rather than type ID. `Is("...ID..")` will filter by type ID. But in this example, yes, they will produce the same result.

epihacker commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/ActivityStreams/PageSubscriptionRepository.cs:
```diff
@@ -167,17 +165,21 @@ private IEnumerable<PageSubscription> AdaptSocialSubscription(List<Subscription>
+
+            var targetFilter = this.subscriptionFilters.Target.EqualTo(
+                !string.IsNullOrWhiteSpace(filter.Target) ? filter.Target : ""
+            );
```
for the conditional return to work properly filters must be initiated as:

var filters = new List<ListExpression>();

epihacker commented on src/EPiServer.SocialAlloy.ExtensionData/EPiServer.SocialAlloy.ExtensionData.csproj:
```diff
@@ -30,36 +30,36 @@
-    <Reference Include="EPiServer.Social.Common, Version=1.5.0.0, Culture=neutral, PublicKeyToken=8fe83dea738b45b7, processorArchitecture=MSIL">
-      <HintPath>..\packages\EPiServer.Social.Common.1.5.0\lib\EPiServer.Social.Common.dll</HintPath>
+    <Reference Include="EPiServer.Social.Common, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8fe83dea738b45b7, processorArchitecture=MSIL">
+      <HintPath>..\packages\EPiServer.Social.Common.2.0.0-pre-000802\lib\EPiServer.Social.Common.dll</HintPath>
```
agreed to do a final cleanup of this after 2.0 is released.

epihacker commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/ActivityStreams/PageSubscriptionRepository.cs:
```diff
@@ -104,21 +105,17 @@ public bool Exist(PageSubscriptionFilter filter)
-                );
+                var removeSubscription = AdaptSubscription(subscription);
+
+                var subscriptionFilters = new SubscriptionFilters();
```
good catch!

epihacker commented on src/EPiServer.SocialAlloy.Web/Social/Extensions/CommentExtensions.cs:
```diff
@@ -38,18 +41,15 @@ static CommentExtensions()
         /// Returns a page of comments.
         /// </returns> 
-        public static ResultPage<Comment> GetComments(this IContent content, Visibility visibile, int offset, int size)
+        public static ResultPage<Comment> GetComments(this IContent content, bool visibile, int offset, int size)
```
yeah, cool...will rework

epihacker commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Moderation/CommunityMembershipModerationRepository.cs:
```diff
@@ -355,27 +356,27 @@ public void Moderate(string workflowId, string action, string userId, string com
-                if (listOfWorkflow.Count() > 0)
+                IEnumerable<Workflow<MembershipModeration>> workflows = Enumerable.Empty<Workflow<MembershipModeration>>();
+
+                var workflowFilters = new FieldExpressionBuilder<MembershipModeration>();
```
I thought <Feature>Filters are only used for core model filtering?  The filters applied here are for extension data so I think I need to use a FieldExpressionFilder as currently done?

epishilpa replied:
If I understand correctly, Chris was probably suggesting this:
var workflowFilters = new WorkflowFilters();
var filterWorkflowsByGroup = workflowFilters.Extension.Field<MembershipModeration>(w => w.Group).EqualTo(group);

epishilpa replied:
.Field is followed by MembershipModeration in angle brackets. The editor is hiding the angle brackets, maybe i should have escaped those while posting comment.
var filterWorkflowsByGroup = workflowFilters.Extension.Field&lt;MembershipModeration&gt;(w => w.Group).EqualTo(group);

epihacker replied:
ah crap...forgot to come back to this one...yeah, you're right...much simpler this way :-)

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Models/Ratings/RatingBlockViewModel.cs:
```diff
@@ -91,11 +91,11 @@ private void LoadRatingSettings(RatingBlock block)
-
-            RatingSettings.AddRange(block.RatingSettings.Select(r => r.Value));
-            RatingSettings.Sort();
+            if (block.RatingSettings?.Any() == true)
```
Minor: maybe remove the == true here?

epihacker replied:
The way it's being used with the nullable you need an explicit comparison.

epihacker commented:
@cbanner I think I've addressed all your comments.  Also made some extra tweaks to demonstrate more of the 2.0 features (e.g., introduced TypeId filtering for feed items).

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Models/ActivityStreams/PageCommentActivity.cs:
```diff
@@ -1,24 +1,21 @@
 {
     /// <summary>
-    /// Represents the activity of posting a comment in the Episerver Social sample.
+    /// Represents the activity of a page comment being submitted in the site.
```
maybe: the activity of a comment being submitted on a page in the site?

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Models/ActivityStreams/PageRatingActivity.cs:
```diff
@@ -1,24 +1,21 @@
 {
     /// <summary>
-    /// Represents the activity of a user rating being submitted in the site.
+    /// Represents the activity of a page rating being submitted in the site.
```
maybe: the activity of a rating being submitted on a page in the site.
or the activity of a page being rated by a user navigating the site?

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Models/ActivityStreams/PageRatingActivity.cs:
```diff
@@ -1,24 +1,21 @@
+        /// The extension data class TypeId.
         /// </summary>
-        public int Value { get; set; }
+        public const string TypeId = "bf41f9e2-50f1-4ee4-bed3-e03f1914f720";
```
Do you think we should use strings or Guids for TypeId. I think the flexibility to have either is quite powerful. Or maybe a comment explaining what TypeId is and the possible type of values it could have, since it is new with 2.0?

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Groups/CommunityMemberRepository.cs:
```diff
@@ -100,41 +96,40 @@ public IEnumerable<CommunityMember> Get(CommunityMemberFilter communityMemberFil
         /// <summary>
-        /// Build the appropriate CompositeCriteria based the provided CommunityMemberFilter.
-        /// The member filter will either contain a group id or a logged in user id. If neitheris provided an exception is thrown.
+        /// Build the appropriate criteria based the provided CommunityMemberFilter.
```
based on?

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Groups/CommunityMemberRepository.cs:
```diff
@@ -100,41 +96,40 @@ public IEnumerable<CommunityMember> Get(CommunityMemberFilter communityMemberFil
+        /// If neither is provided an exception is thrown.
         /// </summary>
         /// <param name="communityMemberFilter">The provided member filter</param>
         /// <returns>A composite criteria of type MemberFilter and MemberExtensionData</returns>
```
<returns> need to be updated to say Criteria.

epishilpa commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Moderation/CommunityMembershipModerationRepository.cs:
```diff
@@ -127,18 +131,23 @@ private AddMemberRequest GetMembershipRequest(string userId, string communityId)
 
             try
             {
                 //retrieve the first workflow that matches the target filter 
```
Minor: Please add a space after // and capitalize first letter in "retrieve" to match our comment styling elsewhere. thanks.

cbanner commented on src/EPiServer.SocialAlloy.Web/Social/Repositories/Ratings/PageRatingRepository.cs:
```diff
@@ -76,24 +80,35 @@ public void AddRating(string user, string target, int value)
+
+            if (!string.IsNullOrWhiteSpace(filter.Rater))
+            {
+                this.ratingFilters.Rater.EqualTo(Reference.Create(filter.Rater));
```
The filters are now being conditionally created but they aren't being assigned to anything. Need a filters.Add(...) around the expressions being created here.

epihacker replied:
Duh...:-)  Shouldn't the AndExpression have complained in that case?

cbanner commented on src/EPiServer.SocialAlloy.Web/Social/Extensions/CommentExtensions.cs:
```diff
@@ -25,31 +27,36 @@ public static class CommentExtensions
-                    Parent = targetReference,
-                    Visibility = visibile
-                },
+                filters.Add(commentFilters.IsVisible.EqualTo(visible == true ? true : false));
```
Nullable types expose a `.Value` property to make this sort of thing a little simpler.
`...IsVisible.EqualTo(visible.Value)`
